### PR TITLE
Andrew hou/fd 106503/update linter for url field

### PR DESCRIPTION
**Pull request by andrewwhou**
Updated `StringsLint` to parse the new `img` field of the comment section.

### CODE REVIEW

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift [hunk 1]
```diff
@@ -87,12 +87,16 @@ extension JSONCommentRule: LintRule {
 
         DispatchQueue.concurrentPerform(iterations: self.declaredStrings.count) { index in
             let string = self.declaredStrings[index]
+            var severity = self.severity
             if let violationType = getViolation(for: string) {
                 lock.lock()
+                if ViolationType.missingScreenshotURL.reasonDescription == violationType.reasonDescription || ViolationType.invalidScreenshotURL.reasonDescription == violationType.reasonDescription {
```
Comment (harkmall) [inline Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift:93]: I think what would make more sense here and would be more robust going forward is to introduce a `violationSeverityOverride` value to the `ViolationType` enum that we can define an override value for for each different violation. Maybe something like this:
```swift
var severityOverride: ViolationSeverity? {
  switch self {
  case .invalidJSON:
    return nil
    <etc>
  }
}
```

and then in building the `Violation` type below, we can do something like:

```swift
severity: violation.severityOverride ?? self.severity
```

**Review by harkmall — CHANGES_REQUESTED** [review]

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift (hunk 1 above)
Comment (andrewwhou) [reply Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift]: Yep this makes a lot more sense. Just pushed

File: Sources/StringsLintFramework/Models/LocalizedString.swift [hunk 2]
```diff
@@ -12,6 +12,7 @@ public struct LocalizedString: CustomStringConvertible, Hashable {
     let location: Location
     var placeholders: [String] = []
     var comment: String? = nil
+    var value: [String]? = nil
```
Comment (harkmall) [inline Sources/StringsLintFramework/Models/LocalizedString.swift:15]: Why is the value an array?

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift [hunk 3]
```diff
@@ -198,11 +212,13 @@ extension JSONCommentRule {
         let descriptionString: String?
         let placeholders: [String]?
         let img: String?
+        let max_character_count: Int?
```
Comment (harkmall) [inline Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift:220]: We try and use snakeCase for variable names.
```suggestion
        let maxCharacterCount: Int?
```

**Review by harkmall — CHANGES_REQUESTED** [review]

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift (hunk 3 above)
Comment (andrewwhou) [reply Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift:220]: I saw Sebastian's using that on Android, so we don't have to match exactly?

File: Sources/StringsLintFramework/Models/LocalizedString.swift (hunk 2 above)
Comment (andrewwhou) [reply Sources/StringsLintFramework/Models/LocalizedString.swift:15]: I was assuming we need to make sure every possible string in `StringsDict` also don't exceed the length?

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift (hunk 3 above)
Comment (harkmall) [reply Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift:220]: we don't have to match, no

File: Sources/StringsLintFramework/Models/LocalizedString.swift (hunk 2 above)
Comment (harkmall) [reply Sources/StringsLintFramework/Models/LocalizedString.swift:15]: Ah yes, good call. I missed that.

File: Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift (hunk 3 above)
Comment (harkmall) [reply Sources/StringsLintFramework/Rules/Lint/JSONCommentRule.swift:220]: we can have the JSON in the comment be snake_case but the variable we use in the swift linter code should be camelCase

**Review by harkmall — APPROVED** [review]

**Pull request merged**